src/uwtools/config/tools.py

```diff
@@ -79,33 +78,24 @@ def config_check_depths_update(config_obj: Union[Config, dict], target_format: s
 def realize_config(
     input_config: Union[Config, Optional[Path]] = None,
     input_format: Optional[str] = None,
-    output_block: Optional[List[Union[str, int]]] = None,
+    update_config: Union[Config, Optional[Path]] = None,
+    update_format: Optional[str] = None,
     output_file: Optional[Path] = None,
     output_format: Optional[str] = None,
-    supplemental_configs: Optional[List[Union[dict, Config, Path]]] = None,
+    output_block: Optional[List[Union[str, int]]] = None,
     values_needed: bool = False,
     total: bool = False,
     dry_run: bool = False,
 ) -> dict:
     """
     NB: This docstring is dynamically replaced: See realize_config.__doc__ definition below.
     """
-    input_format = _ensure_format("input", input_format, input_config)
-    output_format = _ensure_format("output", output_format, output_file)
-    _validate_format_output(input_format, output_format)
-    input_obj: Config = (
-        input_config
-        if isinstance(input_config, Config)
-        else format_to_config(input_format)(config=input_config)
-    )
-    if supplemental_configs:
-        input_obj = _realize_config_update(input_obj, input_format, supplemental_configs)
+    input_obj, input_format = _realize_config_input_setup(input_config, input_format)
+    input_obj = _realize_config_update(input_obj, update_config, update_format)
     input_obj.dereference()
-    output_data = input_obj.data
-    if output_block is not None:
-        for key in output_block:
-            output_data = output_data.get(key, {})
-    config_check_depths_realize(output_data, output_format)
+    output_data, output_format = _realize_config_output_setup(
+        input_obj, output_file, output_format, output_block
+    )
```

Comment:
The old code between the function signature and here grew in complexity, so I factored it out to three separate functions -- `_realize_config_input_setup()`, `_realize_config_output_setup()`, and `_realize_config_update()` -- each with their own, new tests.